### ISSUE TITLE
An exception occurred during MKCOL request: [Errno 22] Invalid argument

### ISSUE THREAD

**Issue by bilogic** [Kozea/Radicale#656]
I'm trying Radicale out. Can login, but can't "Create new collection". Keeps returning 500 Internal Server Error. My config file, follow by the debug logs. Hope someone can help. Thank you.
```
[server]
# hosts = 0.0.0.0:5232,192.168.214.10:5232
hosts = 192.168.214.10:5232

[rights]
#type = owner_only
type = none

[storage]
filesystem_folder = /var/www/html/radicale/collections

[auth]
type = htpasswd
htpasswd_filename = /var/www/html/radicale/htpasswd
# encryption method used in the htpasswd file
htpasswd_encryption = plain
```

My command line startup
```
python3 -m radicale -D --config /var/www/html/radicale/config
```

Debug logs
```
[7f364ffff700] INFO: MKCOL request for '/test/fc3fdfb3-0f7a-74a3-5bed-78db8561b076/' received from 192.168.214.83 using 'Mozilla/5.0 (Windows NT 6.2; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36'
[7f364ffff700] DEBUG: Request headers:
{'CONTENT_LENGTH': '634',
 'CONTENT_TYPE': 'text/plain;charset=UTF-8',
 'GATEWAY_INTERFACE': 'CGI/1.1',
 'HTTP_ACCEPT': '*/*',
 'HTTP_ACCEPT_ENCODING': 'gzip, deflate',
 'HTTP_ACCEPT_LANGUAGE': 'en-US,en;q=0.8',
 'HTTP_CONNECTION': 'keep-alive',
 'HTTP_HOST': '192.168.214.10:5232',
 'HTTP_ORIGIN': 'http://192.168.214.10:5232',
 'HTTP_REFERER': 'http://192.168.214.10:5232/.web/',
 'HTTP_USER_AGENT': 'Mozilla/5.0 (Windows NT 6.2; Win64; x64) '
                    'AppleWebKit/537.36 (KHTML, like Gecko) '
                    'Chrome/59.0.3071.115 Safari/537.36',
 'PATH_INFO': '/test/fc3fdfb3-0f7a-74a3-5bed-78db8561b076/',
 'QUERY_STRING': '',
 'REMOTE_ADDR': '192.168.214.83',
 'REMOTE_HOST': '',
 'REQUEST_METHOD': 'MKCOL',
 'SCRIPT_NAME': '',
 'SERVER_NAME': 'LinuxDev.lan',
 'SERVER_PORT': '5232',
 'SERVER_PROTOCOL': 'HTTP/1.1',
 'SERVER_SOFTWARE': 'WSGIServer/0.2',
 'wsgi.errors': <_io.StringIO object at 0x7f3655e2f438>,
 'wsgi.file_wrapper': <class 'wsgiref.util.FileWrapper'>,
 'wsgi.input': <_io.BufferedReader name=7>,
 'wsgi.multiprocess': False,
 'wsgi.multithread': True,
 'wsgi.run_once': False,
 'wsgi.url_scheme': 'http',
 'wsgi.version': (1, 0)}
[7f364ffff700] DEBUG: Sanitized script name: ''
[7f364ffff700] DEBUG: Sanitized path: '/test/fc3fdfb3-0f7a-74a3-5bed-78db8561b076/'
[7f364ffff700] DEBUG: Request content:
<?xml version="1.0"?>
<mkcol xmlns="DAV:" xmlns:C="urn:ietf:params:xml:ns:caldav" xmlns:CR="urn:ietf:params:xml:ns:carddav" xmlns:ICAL="http://apple.com/ns/ical/" xmlns:ns3="http://inf-it.com/ns/ab/">
  <set>
    <prop>
      <resourcetype>
        <collection />
        <C:calendar />
      </resourcetype>
      <C:supported-calendar-component-set>
        <C:comp name="VEVENT" />
      </C:supported-calendar-component-set>
      <displayname>test</displayname>
      <ICAL:calendar-color>#cdeb0aff</ICAL:calendar-color>
      <ns3:addressbook-color />
      <CR:addressbook-description />
      <C:calendar-description>test</C:calendar-description>
    </prop>
  </set>
</mkcol>

[7f364ffff700] ERROR: An exception occurred during MKCOL request on '/test/fc3fdfb3-0f7a-74a3-5bed-78db8561b076/': [Errno 22] Invalid argument
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/radicale/__init__.py", line 298, in __call__
    status, headers, answers = self._handle_request(environ)
  File "/usr/lib/python3.4/site-packages/radicale/__init__.py", line 464, in _handle_request
    environ, base_prefix, path, user)
  File "/usr/lib/python3.4/site-packages/radicale/__init__.py", line 650, in do_MKCOL
    self.Collection.create_collection(path, props=props)
  File "/usr/lib/python3.4/site-packages/radicale/storage.py", line 789, in create_collection
    self.set_meta(props)
  File "/usr/lib/python3.4/site-packages/radicale/storage.py", line 1257, in set_meta
    json.dump(new_props, f)
  File "/usr/lib64/python3.4/contextlib.py", line 66, in __exit__
    next(self.gen)
  File "/usr/lib/python3.4/site-packages/radicale/storage.py", line 661, in _atomic_write
    self._sync_directory(directory)
  File "/usr/lib/python3.4/site-packages/radicale/storage.py", line 693, in _sync_directory
    cls._fsync(fd)
  File "/usr/lib/python3.4/site-packages/radicale/storage.py", line 679, in _fsync
    os.fsync(fd)
OSError: [Errno 22] Invalid argument


```

**Comment by Unrud** [comment]
fsync'ing folders seems to fail. As a workaround you can disable it:

```
[storage]
filesystem_fsync = False
```

  * What operating system are you using?
  * What kernel version are you using?
  * What file system are you using at ``/var/www/html/radicale/collections``?

**Comment by bilogic** [comment]
Can I ask, what am I missing by disabling fsync?

- centos-release-6-4.el6.centos.10.x86_64
- its NTFS mapped via samba ```/var/www/html/radicale/collections```

**Comment by Unrud** [comment]
> Can I ask, what am I missing by disabling fsync?

From the documentation:

> Disabling it increases the risk of data loss, when the system crashes or power fails!

I tested it on Fedora 25 and it works for me. I guess that the problem got fixed in a newer version of Samba.

**Comment by Unrud** [comment]
Fixed by d3a90d16c3548869d3ca21147d983bb733b12d1f
